libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -282,6 +330,39 @@ bool _IotMqtt_ValidatePublish( bool awsIotMqttMode,
 
 /*-----------------------------------------------------------*/
 
+bool _IotMqtt_ValidatePublish( bool awsIotMqttMode,
```

Comment:
Only one function for validating publishes is needed.  The first parameter is `awsIotMqttMode` anyways, so the maximum payload length check should be done there.

Reply:
That doesn't handle the LWT case, which requires a different maximum length.  It is true that we could modify `_IotMqtt_ValidatePublish()` by adding a third parameter `IsLwtTopic`, and then modifying its logic with additional tests for LWT to select the appropriate size.  This would make calls to `_IotMqtt_ValidatePublish()` less readable in the publish/receive code, which is why the additional function `_IotMqtt_ValidateLwtPublish()` was added for the connect case instead.